File: code_combat/8_Cloudrip_Mountain/399-Pesky_Yaks/pesky_yaks.py
def removeByType(enemies, excludedType):
    tempArray = []
    for enemy in enemies:
        if enemy.type != excludedType:
            tempArray.append(enemy)
    return tempArray


while True:
    enemy = hero.findNearest(removeByType(hero.findEnemies(), "sand-yak"))
    if enemy:
        hero.attack(enemy)

File: projects/code_combat/8_Cloudrip_Mountain/399-Pesky_Yaks/peskyYaks.js
// That's a lot of Yaks!
// If you are to survive, you'll need to filter out Yaks...

function removeByType(enemies, excludedType) {
    var tempArray = [];
    // Go through each enemy and check to see if its type is excludedType.
    for(var i = 0; i < enemies.length; i++) {
        // If it isn't, 'push' it onto the array.
        var enemy = enemies[i];
        if (enemy.type != excludedType)
            tempArray.push(enemy);
    }
    return tempArray;
}


while(true) {
    // Find the enemies!
    var enemies = hero.findEnemies();
    // Remove those pesky Yaks.
    enemies = removeByType(enemies, "sand-yak");
    var enemy = hero.findNearest(enemies);
    
    // Now... 'remove' those enemies.
    if(enemy) {
        hero.attack(enemy);
    }
}
